tests/run_test.go

```diff
@@ -0,0 +1,217 @@
+
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh -c "{{.svc1Cmd}}"
+        startup: enabled
+`
+	svc1Cmd := fmt.Sprintf("touch %s ; sleep 1000", filepath.Join(pebbleDir, "svc1"))
+	layerYAML = strings.Replace(layerYAML, "{{.svc1Cmd}}", svc1Cmd, -1)
+
+	createLayer(t, pebbleDir, "001-simple-layer.yaml", layerYAML)
+
+	logsCh := pebbleRun(t, pebbleDir, "--hold")
+	expectedLogs := []string{"Started daemon"}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+
+	// Sleep a second before checking services because immediate check
+	// can't guarantee that svc1 is not started shortly after the log "Started daemon".
+	time.Sleep(time.Second)
+
+	_, err := os.Stat(filepath.Join(pebbleDir, "svc1"))
+	if err == nil {
+		t.Error("pebble run --hold failed, services are still started")
+	} else {
+		if !os.IsNotExist(err) {
+			t.Errorf("Error checking service %s: %v", "svc1", err)
+			fmt.Printf("Error checking the file: %v\n", err)
+		}
+	}
+}
+
+func TestHttpPort(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	port := "4000"
+	logsCh := pebbleRun(t, pebbleDir, "--http=:"+port)
+	expectedLogs := []string{"Started daemon"}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+
+	if !isPortUsedByProcess(t, port, "pebble") {
+		t.Errorf("Pebble is not listening on port %s", port)
+	}
+}
+
+func TestVerbose(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh -c "{{.svc1Cmd}}"
+        startup: enabled
+`
+	layersFileName := "001-simple-layer.yaml"
+	svc1Cmd := fmt.Sprintf("cat %s; sleep 1000", filepath.Join(pebbleDir, "layers", layersFileName))
+	layerYAML = strings.Replace(layerYAML, "{{.svc1Cmd}}", svc1Cmd, -1)
+
+	createLayer(t, pebbleDir, layersFileName, layerYAML)
+
+	logsCh := pebbleRun(t, pebbleDir, "--verbose")
+	expectedLogs := []string{
+		"Started daemon",
+		"services:",
+		"svc1:",
+		"override: replace",
+		"startup: enabled",
+		"command: /bin/sh -c",
+	}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+}
+
+func TestArgs(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh
+        startup: enabled
+`
+	layersFileName := "001-simple-layer.yaml"
+	svc1Cmd := fmt.Sprintf("cat %s; sleep 1000", filepath.Join(pebbleDir, "layers", layersFileName))
+	layerYAML = strings.Replace(layerYAML, "{{.svc1Cmd}}", svc1Cmd, -1)
+
+	createLayer(t, pebbleDir, layersFileName, layerYAML)
+
+	logsCh := pebbleRun(t, pebbleDir, "--verbose",
+		"--args",
+		"svc1",
+		"-c",
+		fmt.Sprintf("cat %s; sleep 1000", filepath.Join(pebbleDir, "layers", layersFileName)),
+	)
+	expectedLogs := []string{
+		"Started daemon",
+		"services:",
```

Comment:
Similar comments here as above: can we just have it output a fixed string, rather than printing its own layers file?